pkg/client/combined/combined.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2019 The GitLab-Controller Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package combined
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Client is a client.Client that updates the metadata, spec, and status of
+// objects using the status subresource with one Update call. All other calls
+// are passed directly to the underlying client.Client. Client should only be
+// used with resources that have +kubebuilder:subresource:status enabled.
+type Client struct {
```

Comment:
I don't think I am "on board" with this. We should either disable "status" sub-resource or use it as is.